api/resources/providers/gcp/gcp.go

```diff
@@ -56,3 +70,96 @@ func (*Client) NewMetricsClient(ctx context.Context, opts ...option.ClientOption
 
 	return &metric.Client{MetricClient: mCl}, nil
 }
+
+// Implement methods matching the CloudResourceProvider interface.
+func (c *Client) ListResources(ctx *gofr.Context, creds any, filter models.ResourceFilter) ([]models.Resource, error) {
+	var allResources []models.Resource
+
+	resourceTypes := filter.ResourceTypes
+
+	includeSQL := len(resourceTypes) == 0
+
+	for _, t := range resourceTypes {
+		if t == databaseSQL || t == allResource {
+			includeSQL = true
+		}
+	}
+
+	if includeSQL {
+		credsObj, err := c.NewGoogleCredentials(ctx, creds, sqladmin.SqlserviceAdminScope)
+		if err != nil {
+			return nil, ErrInvalidCredentials
+		}
+
+		sqlClient, err := c.NewSQLClient(ctx, option.WithCredentials(credsObj))
+		if err != nil {
+			return nil, ErrInitializingClient
+		}
+		// Extract projectID from creds
+		var credStruct struct{ ProjectID string }
+
+		b, _ := json.Marshal(creds)
+		_ = json.Unmarshal(b, &credStruct)
+
+		instances, err := sqlClient.GetAllInstances(ctx, credStruct.ProjectID)
```

Comment:
We can use the google credential object(credObj - can be named better) to get the ProjectID